engine/common/requester/engine.go

```diff
@@ -234,19 +237,26 @@ func (e *Engine) addEntityRequest(entityID flow.Identifier, selector flow.Identi
 // Force will force the requester engine to dispatch all currently
 // valid batch requests.
 func (e *Engine) Force() {
-	count := uint(0)
-	for {
-		dispatched, err := e.dispatchRequest()
-		if err != nil {
-			e.log.Error().Err(err).Msg("could not dispatch requests")
-			return
-		}
-		if !dispatched {
-			e.log.Debug().Uint("requests", count).Msg("forced request dispatch")
-			return
+	// using Launch to ensure the caller won't be blocked
```

Comment:
the performance problems seem to indicate that the `Force` might run for some time (?) If that is the case, we could include some logic for exiting early, because the `unit.Launch` has some overhead attached to it:

```suggestion
	// exit early in case a forced dispatch is currently ongoing
	if e.isDispatching.Load() { 
	   return
	}
	// using Launch to ensure the caller won't be blocked
```